electron/ProcessingHelper.ts

```diff
@@ -75,7 +88,7 @@ export class ProcessingHelper {
           this.geminiApiKey = null;
           console.warn("No API key available, OpenAI client not initialized");
         }
-      } else {
+      } else if (config.apiProvider === "gemini"){
         // Gemini client initialization
         this.openaiClient = null;
```

Comment:
All these values that we set should be consistently set throughout.
Shouldn't we also add it for anthropic here.
```suggestion
        this.openaiClient = null;
        this.anthropicClient = null;
```
Please add wherever missing for each of the 3 providers such that for each the other 2 are null